src/ext/filer.lisp

```diff
@@ -190,3 +191,21 @@
 
 (define-command filer-select () ()
   (select (back-to-indentation (current-point))))
+
+(define-command filer-delete-select () ()
+  "Delete the selected file in the filer."
+  (let ((item (text-property-at (back-to-indentation (current-point)) :item)))
+    (unless item
+      (editor-error "No file selected")
+      (return-from filer-delete-select))
+    (let ((file (item-pathname item)))
+      (when (and file
+                 (prompt-for-y-or-n-p (format nil "Do you really want to delete \"~A\"?" file)))
+        (handler-case
+            (lem/directory-mode/file:delete-file* file)
+          (sb-ext:delete-file-error (e)
```

Comment:
I suggest adding a `#+sbcl` feature flag: even though SBCL is the main implementation we use, there has been work to run Lem on others.

(and so have a `#-sbcl` for other impls too)